### PR TITLE
fix(coinbase): Fixed coinbase v2 API

### PR DESCRIPTION
**Pull request by Igoranze**
# Submitting Pull Requests

## General

 - [ ] Make sure you use [semantic commit messages](https://seesparkbox.com/foundry/semantic_commit_messages).
       Examples: `"fix(google): Fixed foobar bug"`, `"feat(accounts): Added foobar feature"`.
 - [ ] All Python code must formatted using Black, and clean from pep8 and isort issues.
 - [ ] JavaScript code should adhere to [StandardJS](https://standardjs.com).
 - [ ] If your changes are significant, please update `ChangeLog.rst`.
 - [ ] If your change is substantial, feel free to add yourself to `AUTHORS`.

 ## Provider Specifics

 In case you add a new provider:

- [ ] Make sure unit tests are available.
- [ ] Add an entry of your provider in `test_settings.py::INSTALLED_APPS` and `docs/installation.rst::INSTALLED_APPS`.
- [ ] Add documentation to `docs/providers.rst`.
- [ ] Add an entry to the list of supported providers over at `docs/overview.rst`.

### CODE REVIEW

**Comment by Igoranze** [comment]
Made a PR with the latest build. This was the previous PR that can be omitted: #1898 (https://github.com/pennersr/django-allauth/pull/1898)

**Comment by Igoranze** [comment]
@pennersr do i need to improve more?
I think this should be sufficient

File: allauth/socialaccount/providers/coinbase/provider.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ class CoinbaseProvider(OAuth2Provider):
 
     def get_default_scope(self):
         # See: https://coinbase.com/docs/api/permissions
-        return ["user"]
+        return ["wallets:addresses:read", "wallet:user:email"]
```
Comment (pennersr) [inline allauth/socialaccount/providers/coinbase/provider.py:22]: **Issue**: For allauth purposes, we do not need `wallets:addresses:read`. I would prefer for the default scope to be conservative and only include the email

**Review by pennersr — CHANGES_REQUESTED** [review]

**Comment by pennersr** [comment]
> @pennersr do i need to improve more? I think this should be sufficient

This PR breaks the build...

File: allauth/socialaccount/providers/coinbase/provider.py (hunk 1 above)
Comment (Igoranze) [reply allauth/socialaccount/providers/coinbase/provider.py:22]: I have fixed this issue locally, need to update the PR.
But I notice when I run the test locally that a lot of providers do not have a mock response in their test.py.
What do we need to do with those?

I think that should be another PR, how about you?

**Comment by Igoranze** [comment]
@pennersr should be alright now

**Comment by Igoranze** [comment]
👍 Nice!

**Pull request merged**